### PR TITLE
add support to run tests with different user permissions

### PR DESCRIPTION
**Pull request by christianvogt**
https://issues.redhat.com/browse/RHOAIENG-5259

## Description

Added utility to run tests as users with various permissions:
- disallowed user: no access to load dashboard
- cluster admin: full access to cluster
- project admin: full access specific projects
- edit user: full access to specific projects without ability to create role bindings

Optionally a user may also be a self provisioner (true by default). Non self provisioners cannot create new projects.

Default user for all tests is now a project admin.

Added some small tests that use the various users as well as updated breaking tests.

## How Has This Been Tested?

Run cypress tests.

## Test Impact

N/A

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/__tests__/cypress/cypress/e2e/Application.cy.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  });
+
+  it('should not show Settings nav section for non-cluster admins', () => {
+    asProjectAdminUser();
+    appChrome.visit();
+    appChrome.findNavSection('Settings').should('not.exist');
+  });
+
+  it('should show Settings nav section for cluster admins', () => {
+    asClusterAdminUser();
```
Comment (andrewballantyne) [inline frontend/src/__tests__/cypress/cypress/e2e/Application.cy.ts:23]: Can I suggest we either have a test for "cluster admin" or "product admin" -- Cluster Admin sounds like `cluster-admin` and that's technically different than just the admins of our product.

File: frontend/src/__tests__/cypress/cypress/utils/users.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ */
+export const asDisallowedUser = (): void => {
+  cy.interceptOdh('GET /api/status', mockStatus({ isAllowed: false }));
+  interceptSelfSubjectAccessReview();
+};
+
+/**
+ * Cluster admins have access to perform any action on resource in every project.
+ */
+export const asClusterAdminUser = (): void => {
```
Comment (andrewballantyne) [inline frontend/src/__tests__/cypress/cypress/utils/users.ts:48]: Ah I see -- you're aiming explicitly at `cluster-admin` -- recommend adding `asProductAdmin` as well. Which should not grant access to any project but will the settings stuff. Not sure if that's possible with this current infra atm.

File: frontend/src/__tests__/cypress/cypress/e2e/Application.cy.ts (hunk 1 above)
Comment (christianvogt) [reply frontend/src/__tests__/cypress/cypress/e2e/Application.cy.ts:23]: ah yes. 
Having a distinct difference between product admin vs cluster admin is key. In fact almost all usage of cluster admin I did should be product admin.

File: frontend/src/__tests__/cypress/cypress/utils/users.ts (hunk 2 above)
Comment (christianvogt) [reply frontend/src/__tests__/cypress/cypress/utils/users.ts:48]: I will add `asProductAdmin` as suggested.

**Comment by christianvogt** [comment]
Rebased.
Added user `asProductAdminUser`.
Moved core intercept initialization for k8s objects from `e2e.ts` to `users.ts`. Expectation now is that tests must establish a user before applying their own test specific intercepts as default intercepts are user specific. `e2e.ts` will still establish the default user as a project admin.

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**